client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { merge } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { useFakeTimers } from 'test/helpers/use-sinon';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { retryOnFailure as rof } from '../';
+import { noRetry, exponentialBackoff } from '../policies';
+
+const retryOnFailure = rof();
+const retryWithDelay = delay => rof( () => delay );
+
+const nextError = { fail: 'failed big time' };
+
+const getSites = deepFreeze( http( {
+	method: 'GET',
+	path: '/sites',
+	apiVersion: 'v1',
+} ) );
+
+const withRetries = retryCount => actionOrInbound =>
```

Comment:
I understand the concept of what this function should be doing, and I get the case where originalRequest is defined, I don't understand the first case though 😶